packages/lit-dev-content/site/tutorials/content/carousel/07.md

```diff
@@ -0,0 +1,36 @@
+Now we're ready to position the elements. We use ReactiveElement's `hasUpdated`
+property to avoid animating the first render, and we move only if selection
+has changed (`this.selected !== this.previous`). We go to the right based on
+the `advancing` property we setup in the `selected` accessor. Then calculate
+the move `delta` and add that to the tracking `left` property. It stores a
+`%` value indicating where the slot "container" element should be positioned.
+It's stored as state so the element re-renders correctly if it updates and
+the `selected` property has not changed.
+
+Finally, update the positions based on the `left` property. The animating
+container (`animateLeft`) is the position we've stored in `left`.
+We want the selected item (`selectedLeft`) to be showing so we just negate
+the `left` on the container so it stays in view. And for the previous item
+(`previousLeft`), we place it to the left or right of the selected item based
+on the move `delta`.
+
+```ts
+render() {
+  const shouldMove = this.hasUpdated &&
+      this.selected !== this.previous;
+  const shouldAdvance = shouldMove && this.advancing;
+  const delta = (shouldMove ? Number(shouldAdvance) || -1 : 0) * 100;
+  this.left -= delta;
+  const animateLeft = `${this.left}%`;
+  const selectedLeft = `${-this.left}%`;
+  const previousLeft = `${-this.left - delta}%`;
+```
+
+If we've done that correctly, nothing should change when we click and
+shift-click the element to change the selection.
+
+<aside class="info">
+  Feel free to open the browser's developer tools and inspect the carousel's
+  DOM to see that the style values for `left` are in fact changing in the
+  three spots we set them.
```

Comment:
Maybe also worth noting that if you look behind the scenes of `overflow: hidden`, it's now possible to see the next and prior slide updating as you click - which is very cool!